packages/preview-middleware-client/src/flp/bootstrap.js

```diff
@@ -15,10 +15,41 @@ async function ushellBootstrap(fnCallback) {
         const response = await fetch(`${basePath}/resources/sap-ui-version.json`);
         const json = await response.json();
         const version = json?.libraries?.find((lib) => lib.name === 'sap.ui.core')?.version ?? '1.121.0';
-        const majorUi5Version = parseInt(version.split('.')[0], 10);
+        const [major, minor] = version.split('.');
+        const majorUi5Version = parseInt(major, 10);
+        const minorUi5Version = parseInt(minor, 10);
         if (majorUi5Version >= 2) {
             src = `${basePath}/resources/sap/ushell/bootstrap/sandbox2.js`;
         }
+
+        // For UI5 version 1.71 and below, we need to remove the asyncHints.requests to load the changes in an Adaptation project.
+        // This logic needs to be executed here to have a reliable check for UI5 version and remove the asyncHints.requests before the sandbox is loaded.
+        // The sandbox shell modifies the `window['sap-ushell-config']`.
+        if (majorUi5Version === 1 && minorUi5Version < 72) {
+            const setNestedProperty = (obj) => {
+                if (!obj || typeof obj !== 'object') return;
+
+                const stack = [obj];
+
+                while (stack.length > 0) {
+                    const current = stack.pop();
+
+                    if (current.asyncHints) {
+                        if (current.asyncHints.requests) {
+                            current.asyncHints.requests = [];
+                        }
+                        return;
+                    }
+
+                    for (const key in current) {
+                        if (typeof current[key] === 'object' && current[key] !== null) {
+                            stack.push(current[key]);
+                        }
+                    }
+                }
+            };
+            setNestedProperty(window['sap-ushell-config']['applications']);
+        }
```

Comment:
Can we please have an own function for this and put the comment about what it does and why it's needed in the respective JSDoc? Including `setNestedProperty` into the `ushellBootstrap` function disrupts the readability of the code.

```suggestion
        if (majorUi5Version === 1 && minorUi5Version < 72) {
            removeAsyncHintsRequests();
        }
```

```js
/**
 * For UI5 version 1.71 and below, we need to remove the asyncHints.requests
 * to load the changes in an Adaptation project.
 * This logic needs to be executed here to have a reliable check for
 * UI5 version and remove the asyncHints.requests before the sandbox is loaded.
 * The sandbox shell modifies the `window['sap-ushell-config']`.
 */
function removeAsyncHintsRequests() {
                const obj = window['sap-ushell-config']['applications'];

                if (!obj || typeof obj !== 'object') return;

                const stack = [obj];

                while (stack.length > 0) {
                    const current = stack.pop();

                    if (current.asyncHints) {
                        if (current.asyncHints.requests) {
                            current.asyncHints.requests = [];
                        }
                        return;
                    }

                    for (const key in current) {
                        if (typeof current[key] === 'object' && current[key] !== null) {
                            stack.push(current[key]);
                        }
                    }
                }
}
```

Reply:
Fixed with: 819600b123158267c9a7d86f6e5a259223b10b8c